modules/ts/include/opencv2/ts.hpp

```diff
@@ -123,6 +123,9 @@
 //#pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wsuggest-override"
 #endif
+#if defined(__clang__)
```

Comment:
We should have here also:
- `__APPLE__`
- and compiler version check

Use this to check available macros: https://stackoverflow.com/questions/2224334/gcc-dump-preprocessor-defines

BTW, there is gtest patch for that (but it is large): https://github.com/google/googletest/pull/2815


---

Keep code consistent and use `__OPENCV_BUILD` guard.